client/sites/tools/logs/index.tsx

```diff
@@ -1,253 +1,304 @@
+import page from '@automattic/calypso-router';
+import {
+	Button,
+	__experimentalToggleGroupControl as ToggleGroupControl,
+	__experimentalToggleGroupControlOption as ToggleGroupControlOption,
+	ToggleControl,
+} from '@wordpress/components';
+import { DataViews } from '@wordpress/dataviews';
+import { download } from '@wordpress/icons';
+import { translate } from 'i18n-calypso';
+import { useCallback, useState, useMemo } from 'react';
+import { v4 as uuid } from 'uuid';
+import InlineSupportLink from 'calypso/components/inline-support-link';
 import { useLocalizedMoment } from 'calypso/components/localized-moment';
-import Pagination from 'calypso/components/pagination';
-import { useSiteLogsQuery, FilterType } from 'calypso/data/hosting/use-site-logs-query';
+import NavigationHeader from 'calypso/components/navigation-header';
+import { LogType, PHPLog, ServerLog } from 'calypso/data/hosting/use-site-logs-query';
 import { useInterval } from 'calypso/lib/interval';
-import {
-	getDateRangeQueryParam,
-	updateDateRangeQueryParam,
-	getFilterQueryParam,
-	updateFilterQueryParam,
-} from 'calypso/sites/tools/logs/components/filter-params';
-import { SiteLogsHeader } from 'calypso/sites/tools/logs/components/site-logs-header';
-import { SiteLogsTable } from 'calypso/sites/tools/logs/components/site-logs-table';
-import { SiteLogsToolbar } from 'calypso/sites/tools/logs/components/site-logs-toolbar';
-import { useDispatch, useSelector } from 'calypso/state';
+import { navigate } from 'calypso/lib/navigate';
+import { useSiteLogsDownloader } from 'calypso/sites/tools/logs/hooks/use-site-logs-downloader';
+import { useDispatch } from 'calypso/state';
 import { recordTracksEvent } from 'calypso/state/analytics/actions';
-import { getSelectedSiteId } from 'calypso/state/ui/selectors';
+import { DateTimePicker } from './components/date-time-picker';
+import useActions from './hooks/use-actions';
+import { useCurrentSiteGmtOffset } from './hooks/use-current-site-gmt-offset';
+import useData from './hooks/use-data';
+import useFields from './hooks/use-fields';
+import {
+	default as useView,
+	buildFilter,
+	getSortField,
+	getVisibleFields,
+	getFilterValue,
+} from './hooks/use-view';
+import type { View } from '@wordpress/dataviews';
 import type { Moment } from 'moment';
-
 import './style.scss';
 
-export type LogType = 'php' | 'web';
-
-const DEFAULT_PAGE_SIZE = 50;
-
-export function buildFilterParam(
-	logType: LogType,
-	severity: string,
-	requestType: string,
-	requestStatus: string
-): FilterType {
-	const filters: FilterType = {};
-
-	if ( logType === 'php' ) {
-		if ( severity ) {
-			filters.severity = [ severity ];
-		}
-	}
-
-	if ( logType === 'web' ) {
-		if ( requestType ) {
-			filters.request_type = [ requestType ];
-		}
-		if ( requestStatus ) {
-			filters.status = [ requestStatus ];
-		}
-	}
-
-	return filters;
-}
-
-export const SiteLogs = ( {
+export const SiteLogsDataViews = ( {
```

Comment:
Despite what the diff suggest, SiteLogsDataViews is a brand new implementation based on DataViews, so it's best to ignore the diff.